cli/azd/pkg/compare/compare.go

```diff
@@ -0,0 +1,13 @@
+package compare
+
+import "strings"
+
+// IsStringNilOrEmpty returns true if the string pointer is nil or the trimmed string is empty.
+func IsStringNilOrEmpty(value *string) bool {
+	return value == nil || strings.TrimSpace(*value) == ""
+}
+
+// PtrValueEquals returns true if the pointer is not nil and the value is equal to the expected value.
+func PtrValueEquals[T comparable](actual *T, expected T) bool {
```

Comment:
based on the name, should we get `true` when calling `PrtValueEquals(nil, nil)` ?    If yes, you might want to add this condition. If not, you might want to update the method's name (maybe `PtrWithValueAndEquals`).  

but, really a nit.